src/lib/storage/table.hpp

```diff
@@ -95,11 +95,13 @@ class Table : private Noncopyable {
   /**
    * Creates a new Chunk and appends it to this table.
    * Makes sure the @param segments match with the TableType (only ReferenceSegments or only data containing segments)
-   * En/Disables MVCC for the Chunk depending on whether MVCC is enabled for the table (has_mvcc())
    * This is a convenience method to enable automatically creating a chunk with correct settings given a set of segments.
-   * @param alloc
+   *
+   * @param mvcc_data   If the Table uses MVCC, this can optionally be passed in. If the Table uses MVCC and no
+   *                    mvcc_data is supplied, MvccData will be allocated.
    */
-  void append_chunk(const Segments& segments, const std::optional<PolymorphicAllocator<Chunk>>& alloc = std::nullopt);
+  void append_chunk(const Segments& segments, std::shared_ptr<MvccData> mvcc_data = {},
```

Comment:
```suggestion
  void append_chunk(const Segments& segments, std::shared_ptr<MvccData> mvcc_data = nullptr,
```
I think it is more intuitive